cookbook/security/impersonating_user.rst

```diff
@@ -151,3 +151,49 @@ setting:
                 ),
             ),
         ));
+        
+Events
+------
+
+The firewall dispatches the ``security.switch_user`` event right after the impersonation
+completed. The ``SwitchUserEvent`` is passed to the listener, based on which you are able
```

Comment:
Might be a good idea to refer to the API docs for the `SwitchUserEvent` (using the `class` role).